app/src/page/file/view.js

```diff
@@ -233,6 +282,33 @@ class FilePage extends React.PureComponent {
     return 'file:///' + fileInfo.download_path;
   }
 
+  playerUriForFileInfo = (fileInfo) => {
+    const { streamingUrl } = this.props;
+    if (streamingUrl) {
+      return streamingUrl;
+    }
+    if (this.state.currentStreamUrl) {
+      return this.state.currentStreamUrl;
+    }
+
+    if (fileInfo && fileInfo.download_path) {
+      return this.getEncodedDownloadPath(fileInfo);
+    }
+
+    return null;
+  }
+
+  getEncodedDownloadPath = (fileInfo) => {
```

Comment:
Would it make sense to instead add an `encoded_file_path` property to `fileInfo` in redux? This kind of code seems somewhat out of place in a view component.

Reply:
Apparently, github hid some of the messages by default, so I missed them. The `fileInfo` is returned by the sdk, but I suppose I can create a `makeSelect` selector in order to achieve this.